src/sql/parts/objectExplorer/viewlet/dragAndDropController.ts

```diff
@@ -74,6 +74,11 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 				if (!this._connectionManagementService.canChangeConnectionConfig(source, targetConnectionProfileGroup.id)) {
 					canDragOver = false;
 				}
+			} else if (source instanceof ConnectionProfileGroup) {
+				// Dropping a group to itself or its descendants nodes is not allowed
+				// to avoid creating a circular structure.
+				//
```

Comment:
nit: I don't think we're typically adding the blank `//` line for comments in this code base.  It doesn't bother me, but I'd tend to prefer consistency, other considerations being equal.